package-res/ccc/core/base/chart/chart.axes.js

```diff
@@ -127,16 +141,26 @@ pvc.BaseChart
 
                     AxisClass = this._axisClassByType[type] || pvc.visual.Axis;
                     dataCellsOfTypeByIndex.forEach(function(dataCells) {
-                        var axisIndex = dataCells[0].axisIndex;
-                        new AxisClass(this, type, axisIndex);
-                    }, this);
+                        // CDF603
+                        // Pass the stored state in axis construction
+                        var axisIndex = dataCells[0].axisIndex,
+                            ka = {};
+                        if(oldByType){
```

Comment:
Also, readability could be improved by creating a nested function that would return the previous state, if any.